File: 0141-linked-list-cycle/0141-linked-list-cycle.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public bool HasCycle(ListNode head) {
        HashSet<ListNode> hs = new HashSet<ListNode>();
        ListNode curr = head;
        while(curr != null) {
            if(hs.Contains(curr)) return true;
            hs.Add(curr);
            curr = curr.next;
        }
        return false;
    }
}

File: 0141-linked-list-cycle/0141-linked-list-cycle.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */

/**
 * @param {ListNode} head
 * @return {boolean}
 */
var hasCycle = function(head) {
    let curr = head;
    let st = new Set();
    while(curr) {
        if(st.has(curr)) return true
        st.add(curr);
        curr = curr.next;
    }
    return false;
};